src/H5FDlog.c

```diff
@@ -484,8 +484,8 @@ H5FD__log_open(const char *name, unsigned flags, hid_t fapl_id, haddr_t maxaddr)
 #ifdef H5_HAVE_WIN32_API
     struct _BY_HANDLE_FILE_INFORMATION fileinfo;
 #endif
-    H5_timer_t open_timer = {{0}, {0}, {0}, FALSE}; /* Timer for open() call */
-    H5_timer_t stat_timer = {{0}, {0}, {0}, FALSE}; /* Timer for stat() call */
+    H5_timer_t open_timer; /* Timer for open() call */
+    H5_timer_t stat_timer; /* Timer for stat() call */
```

Comment:
The changes look noisy, but the important changes just get rid of the initialization and move the call to the init function up (the warning is that the {0}s in the initialization don't fully initialize the struct).